### PR TITLE
84 fine tune bert model for the second llm pass

### PR DESCRIPTION
**Pull request by GustawB**
Introduce a new library used for fine-tuning BERT models and integrate it with existing code, as well as use it to prepare pipeline for fine-tuning BERTs for the 2-stage architecture.

### CODE REVIEW

File: src/bert_finetuning/finetuner.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    for token in tokens:
+        if token == "O" or token == "N/A":
+            custom_tokens.append("O")
+        else:
+            custom_tokens.append(f"B-{token}")
+            custom_tokens.append(f"I-{token}")
+    return custom_tokens
+
+
+def __align_labels_with_tokens(labels: list, word_ids: list, bi_split: bool) -> list:
```
Comment (SzymonKozl) [inline src/bert_finetuning/finetuner.py:71]: This function will work properly only in case the unknown label is mapped to `0` - am I right? If yes, this is ok, but please mention it somewhere.

File: src/bert_finetuning/finetuner.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    Function that is responsible for tokenizing the inputs and aligning the labels with the tokens,
+    using __align_labels_with_tokens function. This function is not intended to be used directly.
+
+    param: input_column: The column that contains the inputs
+    param: labels_column: The column that contains the labels
+    param: bi_split: Whether the labels are in B-X and I-X format
+    param: examples: dataset from which the input_column will be tokenized
+    return: The tokenized inputs with aligned labels
+    """
+    tokenized_inputs = __TOKENIZER(
```
Comment (SzymonKozl) [inline src/bert_finetuning/finetuner.py:116]: Maybe we should use __assert_init here if it already exists?

File: src/test/test_finetuner.py
Comment (SzymonKozl) [inline src/test/test_finetuner.py:1]: Why there are no tests for __compute_metrics?

**Review by SzymonKozl — COMMENTED** [review]
GJ, some suggestions and questions from me

File: src/test/test_finetuner.py
Comment (GustawB) [reply src/test/test_finetuner.py:1]: There's not much to test imo. Metrics itself are from outside lib, so testing that is a waste of time. Apart from that, there are two loops with simple logic. I think too simple to test.

File: src/bert_finetuning/finetuner.py (hunk 1 above)
Comment (GustawB) [reply src/bert_finetuning/finetuner.py:72]: Done

File: src/bert_finetuning/finetuner.py (hunk 2 above)
Comment (GustawB) [reply src/bert_finetuning/finetuner.py:118]: Done

File: src/test/test_finetuner.py
Comment (SzymonKozl) [reply src/test/test_finetuner.py:1]: okay, lets close this

**Review by SzymonKozl — APPROVED** [review]
